librocksdb-sys/build.rs

```diff
@@ -288,17 +288,20 @@ fn build_snappy() {
 }
 
 fn try_to_find_and_link_lib(lib_name: &str) -> bool {
+    println!("cargo:rerun-if-env-changed={}_COMPILE", lib_name);
     if let Ok(v) = env::var(&format!("{}_COMPILE", lib_name)) {
         if v.to_lowercase() == "true" || v == "1" {
             return false;
         }
     }
 
+    println!("cargo:rerun-if-env-changed={}_LIB_DIR", lib_name);
+    println!("cargo:rerun-if-env-changed={}_STATIC", lib_name);
     if let Ok(lib_dir) = env::var(&format!("{}_LIB_DIR", lib_name)) {
         println!("cargo:rustc-link-search=native={}", lib_dir);
-        let mode = match env::var_os(&format!("{}_STATIC", lib_name)) {
-            Some(_) => "static",
-            None => "dylib",
+        let mode = match env::var(&format!("{}_STATIC", lib_name)).as_ref().ok().map(|s| s.as_str()) {
+            Some("") | Some("true") | Some("1") | Some("static") => "static",
```

Comment:
1. Does it really make sense to check all these variants? What if I set `ROCKSDB_STATIC=yes`? 
2. Why do you use `env::var`with `.ok()` rather than `env::var_os` ? 
```rust
let mode = match env::var_os(&format!("{}_STATIC", lib_name))
            .as_ref()
            .and_then(|s| s.to_str()) {
...
```
?

Reply:
Will change to `var_os`, I wasn't aware of the internal implementation of https://doc.rust-lang.org/src/std/env.rs.html#236 `fn var`.
Will also add `yes` as an enabling value.

The idea to accept only specific names or empty, is to allow the inverse usage as well: `ROCKSDB_STATIC=false` is something that is intuitive, when `ROCKSDB_STATIC=1` enables `static` linkage.